fdbserver/KeyValueStoreSQLite.actor.cpp

```diff
@@ -93,29 +94,44 @@ struct PageChecksumCodec {
 		SumType sum;
 		SumType *pSumInPage = (SumType *)(pData + dataLen);
 
-		// Write sum directly to page or to sum variable based on mode
-		SumType *sumOut = write ? pSumInPage : &sum;
-		sumOut->part1 = pageNumber; //DO NOT CHANGE
-		sumOut->part2 = 0x5ca1ab1e;
-		hashlittle2(pData, dataLen, &sumOut->part1, &sumOut->part2);
-
-		// Verify if not in write mode
-		if(!write && sum != *pSumInPage) {
-			if(!silent)
-				TraceEvent (SevError, "SQLitePageChecksumFailure")
-					.error(checksum_failed())
-					.detail("CodecPageSize", pageSize)
-					.detail("CodecReserveSize", reserveSize)
-					.detail("Filename", filename)
-					.detail("PageNumber", pageNumber)
-					.detail("PageSize", pageLen)
-					.detail("ChecksumInPage", pSumInPage->toString())
-					.detail("ChecksumCalculated", sum.toString());
-
-			return false;
+		if (write) {
+			// Always write a CRC32 checksum for new pages
+			sum.part1 = 0; // Indicates CRC32 is being used
```

Comment:
There's no need to invoke memcpy here, just change `sum.` to `pSumInPage->` as pSumInPage already points to the destination SumType, hence why memcpy is copying to it.

Reply:
What I was trying to say was you can just change lines 100 and 101 to set pSumInPage->part1 and pSumInPage ->part2 directly.  Since the `if(write)` block always returns there's no reason to use `sum` here, and sum's declaration could actually be moved down.

But this is fine as-is, I am satisfied that memcpy() is gone.